File: EPIJudge/epi_judge_python/gcd.py
def gcd(a, b):
	if not a:
		return b
	if not b:
		return a
	if a & 1 == 0 and b & 1 == 0:
		return gcd(a >> 1, b >> 1) << 1
	elif a & 1 == 1 and b & 1 == 1:
		return gcd(min(a, b), max(a, b) - min(a,b))
	else:
		if a & 1 == 0:
			a >>= 1
		if b & 1 == 0:
			b >>= 1
		return gcd(a, b)

def gcd_iterative(a, b):
	shift_count = 0
	while a and b:
		if a & 1 and b & 1:
			a, b = min(a, b), max(a, b) - min(a, b)
		else: 
			if not a & 1 and not b & 1:
				shift_count += 1
			if not a & 1:
				a >>= 1
			if not b & 1:
				b >>= 1

	return a << shift_count if a else b << shift_count

from sys import exit

from test_framework import generic_test, test_utils

if __name__ == '__main__':
    exit(generic_test.generic_test_main('gcd.tsv', gcd))

File: EPIJudge/epi_judge_cpp/gcd.cc
#include "test_framework/generic_test.h"

using namespace std;

long long Gcd(long long x, long long y) {
  // Implement this placeholder.
	if (x & 1 == 0 && y & 1 == 0) {
		return Gcd(x >> 1, y >> 1) << 1;
	} else if (x & 1 == 1 && y & 1 == 1) {
		return Gcd(min(x, y), max(x, y) - min(x, y));
	} else {
		if (x & 1 == 0) {
			x >>= 1;
		}
		if (y & 1 == 0) {
			y >>= 1;
		}

		return gcd(x, y);
	}
  return 0;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"x", "y"};
  return GenericTestMain(args, "gcd.tsv", &Gcd, DefaultComparator{},
                         param_names);
}
